project-koku/koku
# add check to stop queued tasks from processing

## Jira Ticket

[COST-####](https://issues.redhat.com/browse/COST-####)

## Description

This change will add a check inside the download task to see if a source is disabled and drop out of the task. 

## Testing

1. Checkout Branch
2. Restart Koku
3. load a large data set which has multiple tasks for download
4. See multiple tasks waiting in the queue here http://localhost:5042/api/cost-management/v1/celery_queue_tasks/
5. Change the Unleash flag to disable the source (I just set the flag to True) E.I  This line `res = UNLEASH_CLIENT.is_enabled("cost-management.backend.disable-source", context)` to `res = True`
6. Save the file and watch the worker reload then start dropping the remaining queued tasks.

## Release Notes

```markdown
* [COST-####](https://issues.redhat.com/browse/COST-####) Fix some things
```
